torchvision/prototype/datasets/utils/_internal.py

```diff
@@ -72,8 +72,8 @@ def _getattr_closure(obj: Any, *, attrs: Sequence[str]) -> Any:
     return obj
 
 
-def _path_attribute_accessor(path: pathlib.Path, *, name: str) -> D:
-    return cast(D, _getattr_closure(path, attrs=name.split(".")))
+def _path_attribute_accessor(path: pathlib.Path, *, name: str) -> Any:
+    return _getattr_closure(path, attrs=name.split("."))
```

Comment:
Why is this change needed?
Do you remember why we annotated the return with `D` in the first place?

Reply:
> Why is this change needed?

Same as above. We no longer can use a `TypeVar` as return annotation if we don't have any input with the same type.

> Do you remember why we annotated the return with `D` in the first place?

IIRC, this is a remnant when this function was a local one inside 

https://github.com/pytorch/vision/blob/06d9726d66f97ed8e3d3910da246a52be9c6e168/torchvision/prototype/datasets/utils/_internal.py#L83

In there the annotation was correct and ok, since we have a `D` in the input parameters. At some point we factored them out since `torchdata` doesn't like local functions, and I probably forgot to fix this since `mypy` didn't complain.